src/runtime_src/tools/scripts/service_bundle.sh

```diff
@@ -0,0 +1,236 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (C) 2019-2021 Xilinx, Inc. All rights reserved.
+#
+# This script collects debugging logs for XRT managed sub-systems
+
+TIME=`date +%F-%T`
+ID=`id -un`
+FULL_CMD=${BASH_SOURCE[0]}
+BASE_CMD=`basename "$0"`
+
+progress() {
+	echo -n "running: $func - progress: $(($1+1))/$2"
+	#echo -ne "running: $func - progress: \b\b\b$(($1+1))/$(($2+1))\b\b\b"
+}
+
+usage() {
+    echo "Usage: $BASE_CMD [options]"
+    echo "  options:"
+    echo "  --out, -o           output directory"
+    echo "  --device, -d        card device bdf"
+    echo "  --all, -a           collect all, including large size system core dump"
+    echo "  --verbose, -v       verbose output"
+    echo "Example: $BASE_CMD -d 9f"
+    exit $1
+}
+
+while [ $# -gt 0 ]; do
+	case "$1" in
+		-help)
+			usage 0
+			;;
+		-o | --out)
+			shift
+			PATH_TO_OUTPUT=$1
+			;;
+		-d | --device)
+			shift
+			DEVICE_BDF=$1
+			;;
+		-v | --verbose)
+			VERBOSE=1
+			;;
+		-a | --all)
+			ALL_LOG=1
+			;;
+		* | -* | --*)
+			echo "$1 invalid argument."
+			usage 1
+			;;
+	esac
+	shift
+done
+
+
+#############################
+# Funcs start here
+#############################
+core_dumps()
+{
+	if [ -z $ALL_LOG ];then
+		echo -n " - skipped: -a to enable"
+		return 0
+	fi
+
+	mkdir -p $1/var/crash/
+	cp -r /var/crash/** $1/var/crash/
+	
+	mkdir -p $1/cores/
+	ls *.core > /dev/null 2>&1
+	if [ $? -eq 0 ];then
+		cp *.core $1/cores/
+	fi
+
+	ls /opt/xilinx/xrt/bin/*.core > /dev/null 2>&1
+	if [ $? -eq 0 ];then
+		cp /opt/xilinx/xrt/bin/*.core $1/cores/
+	fi
+}
+
+xbutil_output()
+{
+	xbutil examine -d $DEVICE_BDF:0 > $1/xbutil_examine.txt
+	xbutil examine -d $DEVICE_BDF:0 -r all > $1/xbutil_examine_all.txt
+	xbutil validate -d $DEVICE_BDF:0 -r quick --verbose> $1/xbutil_validate.txt
+}
+
+xbmgmt_output()
+{
+	xbmgmt examine > $1/xbmgmt_examine.txt
+	xbmgmt examine -d $DEVICE_BDF:0 >> $1/xbmgmt_examine.txt
+
+	xbmgmt examine -d $DEVICE_BDF:0 -r all > $1/xbmgmt_examine_all.txt
+}
+
+debug_logs()
+{
+	mkdir -p $1/var/log/
+	cp -r /var/log/** $1/var/log/
+
+	for xclmgmt in `ls /sys/module/xclmgmt/parameters/`
+	do
+		cat /sys/module/xclmgmt/parameters/$xclmgmt > $1/xclmgmt_parameters_$xclmgmt.txt
+	done
+
+	for xocl in `ls /sys/module/xocl/parameters/`
+	do
+		cat /sys/module/xocl/parameters/$xocl > $1/xocl_parameters_$xocl.txt
+	done
+
+	dmesg -T > $1/dmesg_current.txt
+
+	cat /sys/kernel/debug/xclmgmt/trace > $1/sys_kernel_debug_xclmgmt.txt &
+	PID1=$!
+	cat /sys/kernel/debug/xocl/trace > $1/sys_kernel_debug_xocl.txt &
+	PID2=$!
+	sleep 2
```

Comment:
Why the magic sleep.  A comment here would be nice to indicate what is being waited for.

Reply:
sometimes the cat might takes about 1 seconds if there are too many data in the debugfs. I think 2 seconds is safer.